client/landing/stepper/utils/use-snake-cased-keys.ts

```diff
@@ -0,0 +1,12 @@
+import { useMemo } from '@wordpress/element';
+import { convertToSnakeCase } from 'calypso/state/data-layer/utils';
+
+interface Params {
+	input?: Record< string, unknown >;
+}
+
+const useSnakeCasedKeys = ( { input }: Params ) => {
+	return useMemo( () => input && ( convertToSnakeCase( input ) as Params[ 'input' ] ), [ input ] );
```

Comment:
I bet people will pass this function fresh objects every render and the `useMemo` won't be able to function. Maybe we can make a clever hook that memoizes objects by serializing keys and values somehow. Or use KCD's https://github.com/kentcdodds/use-deep-compare-effect.

Reply:
It's definitely a possibility, but I think it sounds like a general issue with Stepper (and more generally) e.g. every single hook that's exposed to flows for implementation is subject to this concern - so is everything built on top to transform. It's almost like we need to wrap everything with such a clever serializer. 

I think there may be value in something like that, but:

1. doing it here feels a bit random i.e. why only this hook? It needs to be planned out. I would also think a provision to "disable" something like that to be of value.
3. it will become a default performance penalty that happens on every update that goes through these hooks. We'd effectively need to "ban" use of `useMemo` and resolve to this default implementation that wraps everything. Why not trust that flow will be smart about it (especially as we go through a phase of minimizing them)? They only need to memoize the edge to "potentially" benefit (as none of this memoizing is or ought to be required)
 
For all I know, memoizing should be an optional behavior, not a default one. Rerenders don't usually translate to UI changes. The fact that `useSnakeCasedKeys` memorizes isn't required, just a provision to allow flows to opt-in when/if needed (as the alternative would mean "input" being memoized or not makes no difference).